sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -250,6 +252,8 @@ class Dataset[T] private[sql](
     val hasMoreData = takeResult.length > numRows
     val data = takeResult.take(numRows)
 
+    lazy val timeZone = TimeZone.getTimeZone(sparkSession.sessionState.conf.sessionLocalTimeZone)
```

Comment:
What happens if a user changes the session timezone? What would be the preferred behavior? Currently `show()` generates the same result every time, but that might be unexpected.